test/scenarios/ynLSD.spec.sol

```diff
@@ -1,78 +1,217 @@
-	// 	Invariants.shareMintIntegrity(currentTotalShares, previousTotalShares, userShares);
-	// 	Invariants.userSharesIntegrity(ynlsd.balanceOf(user), 0, userShares);
-	// }
+		uint256 previousTotalDeposited;
+		uint256 previousTotalAssets;
+		uint256 previousTotalShares;
+
+		address asset = chainAddresses.lsd.WSTETH_ADDRESS;
+
+		// user 1 deposits random1
+		uint256 user1Amount = random1;
+		deal({ token: asset, to: user1, give: user1Amount });
+
+		uint256 user1Shares = ynEigenToken.previewDeposit(IERC20(asset), user1Amount);
+		vm.startPrank(user1);
+		IERC20(asset).approve(address(ynEigenToken), user1Amount);
+		ynEigenToken.deposit(IERC20(asset), user1Amount, user1);
+		vm.stopPrank();
+
+		previousTotalDeposited = 0;
+		previousTotalAssets = 0;
+		previousTotalShares = 0;
+
+		runInvariants(IERC20(asset), user1, previousTotalAssets, previousTotalDeposited, previousTotalShares, user1Amount, user1Shares);
+
+		// user 2 deposits random2
+		uint256 user2Amount = random2;
+		deal({ token: asset, to: user2, give: user2Amount });
+
+		uint256 user2Shares = ynEigenToken.previewDeposit(IERC20(asset), user2Amount);
+		vm.startPrank(user2);
+		IERC20(asset).approve(address(ynEigenToken), user2Amount);
+		ynEigenToken.deposit(IERC20(asset), user2Amount, user2);
+		vm.stopPrank();
+
+		previousTotalDeposited += user1Amount;
+		previousTotalAssets += assetRegistry.convertToUnitOfAccount(IERC20(asset), user1Amount);
+		previousTotalShares += user1Shares;
+
+		runInvariants(IERC20(asset), user2, previousTotalAssets, previousTotalDeposited, previousTotalShares, user2Amount, user2Shares);
+
+		// user 3 deposits random3
+		uint256 user3Amount = random3;
+		deal({ token: asset, to: user3, give: user3Amount });
+
+		uint256 user3Shares = ynEigenToken.previewDeposit(IERC20(asset), user3Amount);
+		vm.startPrank(user3);
+		IERC20(asset).approve(address(ynEigenToken), user3Amount);
+		ynEigenToken.deposit(IERC20(asset), user3Amount, user3);
+		vm.stopPrank();
+
+		previousTotalDeposited += user2Amount;
+		previousTotalAssets += assetRegistry.convertToUnitOfAccount(IERC20(asset), user2Amount);
+		previousTotalShares += user2Shares;
+
+		runInvariants(IERC20(asset), user3, previousTotalAssets, previousTotalDeposited, previousTotalShares, user3Amount, user3Shares);
+	}
+
+	function runInvariants(IERC20 asset, address user, uint256 previousTotalAssets, uint256 previousTotalDeposited, uint256 previousTotalShares, uint256 userAmount, uint256 userShares) public view {
+		Invariants.totalDepositIntegrity(ynEigenToken.assetBalance(asset), previousTotalDeposited, userAmount);
+		Invariants.totalAssetsIntegrity(ynEigenToken.totalAssets(), previousTotalAssets, assetRegistry.convertToUnitOfAccount(asset, userAmount));
+		Invariants.shareMintIntegrity(ynEigenToken.totalSupply(), previousTotalShares, userShares);
+		Invariants.userSharesIntegrity(ynEigenToken.balanceOf(user), 0, userShares);
+	}
+}
+
+contract YnLSDScenarioTest2 is ynEigenIntegrationBaseTest {
```

Comment:
call it YnLSDeScenarioTest2

Reply:
https://github.com/yieldnest/yieldnest-protocol/pull/135/commits/8f49fc673ba5edd9e84829c37635ca9976ed89d9